### PR TITLE
Move internal types of hpa from pkg/apis/extensions to pkg/apis/autoscaling

### PR DESCRIPTION
**Pull request by piosz**
``` release-note
Changed all Kubernetes components to use HPA in version autoscaling/v1 instead of extensions/v1beta1.
HPA in version extensions/v1beta1 is deprecated now and will be removed in Kubernetes 1.4.
```

ref #21577

@lavalamp could you please review or delegate to someone from CSI team?
@janetkuo could you please take a look into the kubelet changes?

cc @fgrzadkowski @jszczepkowski @mwielgus @kubernetes/autoscaling

### CODE REVIEW

**Comment by piosz** [comment]
I'm doing something wrong or there is something broken with clientgen tool. I ran ./hack/update-all.sh and it didn't remove the following files:
```
pkg/client/typed/generated/extensions/unversioned/fake/fake_horizontalpodautoscaler.go
pkg/client/typed/generated/extensions/unversioned/horizontalpodautoscaler.go
```
I removed them manually and it the scripts are still fine with the change.

cc @caesarxuchao

**Comment by wojtek-t** [comment]
I think all our generators don't support "removing" case. Basically, we are looking at types for which we need to generate something and generate that. Same for verification. We are not looking at all what was previously generated and if we need to remove it now. I think that removing is such a rare case that there is not much sense in supporting it, since it won't be trivial.
@lavalamp for opinion also.

**Comment by piosz** [comment]
Thanks @wojtek-t for the explanation. IIUC I did exactly what is expected.

**Comment by caesarxuchao** [comment]
Yeah, wojtek-t is right and you did it right.

**Comment by piosz** [comment]
friendly ping @lavalamp

File: pkg/apis/autoscaling/v1/register.go [hunk 1]
```diff
@@ -32,7 +32,6 @@ var SchemeGroupVersion = unversioned.GroupVersion{Group: GroupName, Version: "v1
 func AddToScheme(scheme *runtime.Scheme) {
 	addKnownTypes(scheme)
 	addDefaultingFuncs(scheme)
-	addConversionFuncs(scheme)
```
Comment (lavalamp) [inline pkg/apis/autoscaling/v1/register.go]: why is this removed?

**Comment by lavalamp** [comment]
LGTM sans one comment. Sorry, I thought I looked at this yesterday, guess not.

File: pkg/apis/autoscaling/v1/register.go (hunk 1 above)
Comment (piosz) [inline pkg/apis/autoscaling/v1/register.go]: I changed internal types to be identical with v1 so that conversions can be auto-generated and we don't have to have our own functions.

**Comment by piosz** [comment]
Thanks @lavalamp for the review.

**Comment by lavalamp** [comment]
lgtm
